<launch>
  <arg name="GROUND_ROBOT_IP" default="$(env ROS_IP)" />
  <arg name="UAV_IP" default="192.168.1.120" />
  <arg name="HIGH_PORT1" default="16484" />
  <arg name="HIGH_PORT2" default="8192" />
  <arg name="LOW_PORT_1" default="1024" />
  <arg name="LOW_PORT_2" default="1025" />
  <arg name="LOW_PORT_3" default="1026" />

  ### GR receiver from UAV
  <node pkg="jsk_network_tools" type="silverhammer_highspeed_receiver.py" name="receiver" output="screen">
    <param name="message" value="cooperation_landing/UAV2GR" />
    <param name="receive_ip" value="$(arg GROUND_ROBOT_IP)" />
    <param name="receive_port" value="$(arg HIGH_PORT1)" />
    <param name="topic_prefix" value="" />
  </node>
  
  ### GR streamer to UAV
  <node pkg="jsk_network_tools" type="silverhammer_highspeed_streamer.py" name="highspeed_streamer" output="screen">
    <param name="message" value="cooperation_landing/GR2UAV" />
    <param name="to_ip" value="$(arg UAV_IP)" />
    <param name="to_port" value="$(arg HIGH_PORT2)" />
    <param name="send_rate" value="5" />
    <param name="bandwidth" value="2000000" />
    <param name="packet_size" value="1000" />
  </node>

  ### GR low speed streamer1 to UAV
  <node pkg="jsk_network_tools" type="silverhammer_lowspeed_streamer.py" name="lowspeed_streamer_takeoff" output="screen">
    <remap from="~input" to="/quadrotor/teleop_command/takeoff" />
    <param name="message" value="std_msgs/Empty" />
    <param name="to_ip" value="$(arg UAV_IP)" />
    <param name="to_port" value="$(arg LOW_PORT_1)" />
    <param name="send_rate" value="1" />
    <param name="event_driven" value="True" />
  </node>

  ### GR low speed streamer2 to UAV
  <node pkg="jsk_network_tools" type="silverhammer_lowspeed_streamer.py" name="lowspeed_streamer_land" output="screen">
    <remap from="~input" to="/quadrotor/teleop_command/land" />
    <param name="message" value="std_msgs/Empty" />
    <param name="to_ip" value="$(arg UAV_IP)" />
    <param name="to_port" value="$(arg LOW_PORT_2)" />
    <param name="send_rate" value="1" />
    <param name="event_driven" value="True" />
  </node>

    ### GR low speed streamer3 to UAV
  <node pkg="jsk_network_tools" type="silverhammer_lowspeed_streamer.py" name="lowspeed_streamer_nav_trigger" output="screen">
    <remap from="~input" to="/quadrotor/uav/nav/trigger" />
    <param name="message" value="std_msgs/Empty" />
    <param name="to_ip" value="$(arg UAV_IP)" />
    <param name="to_port" value="$(arg LOW_PORT_3)" />
    <param name="send_rate" value="1" />
    <param name="event_driven" value="True" />
  </node>

<!--   ### GR low speed streamer3 to UAV -->
<!--   <node pkg="jsk_network_tools" type="silverhammer_lowspeed_streamer.py" name="lowspeed_streamer_FlightNav" output="screen"> -->
<!--     <remap from="~input" to="/quadrotor/uav/nav" /> -->
<!--     <param name="message" value="aerial_robot_msgs/FlightNav" /> -->
<!--     <param name="to_ip" value="$(arg UAV_IP)" /> -->
<!--     <param name="to_port" value="$(arg LOW_PORT_3)" /> -->
<!--     <param name="send_rate" value="1" /> -->
<!--     <param name="event_driven" value="True" /> -->
<!--   </node> -->

</launch>
